platform/o.n.bootstrap/src/org/netbeans/ModuleManager.java

```diff
@@ -1678,10 +1678,12 @@ public boolean hasToEnableCompatModules(Set<Module> modules) throws IllegalArgum
         }
         return false;
     }
-    
+
     private void maybeAddToEnableList(Set<Module> willEnable, Set<Module> mightEnable, Module m, boolean okToFail) {
-        if (! missingDependencies(m).isEmpty()) {
-            assert okToFail : "Module " + m + " had unexpected problems: " + missingDependencies(m) + " (willEnable: " + willEnable + " mightEnable: " + mightEnable + ")";
+        if (!missingDependencies(m).isEmpty()) {
+            if (!okToFail) {
```

Comment:
This seems to change the behaviour for regular modules as well, not just for fragments. Was it intended ?